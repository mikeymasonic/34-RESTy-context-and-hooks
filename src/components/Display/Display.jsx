import React from 'react';
import PropTypes from 'prop-types';
import ReactJson from 'react-json-view';
import styles from './Display.css';

const Display = ({ response, headers }) => {
  // const responseList = Object.keys(response).map(keyValue => {
  //   return <li key={keyValue}>{keyValue}: {response[keyValue]}</li>;
  // });

  return (
    <div className={styles.Display}>
      <ReactJson src={headers} theme="monokai" name="Headers"/>
      <ReactJson src={response} theme="monokai" name="Response" />
    </div>
  );
};

Display.propTypes = {
  headers: PropTypes.object,
  response: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array
  ]),
};

export default Display;
