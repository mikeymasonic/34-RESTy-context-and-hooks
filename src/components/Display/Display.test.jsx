import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('Display component', () => {
  it('renders Display', () => {
    const wrapper = shallow(<Display 
      response={{}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
