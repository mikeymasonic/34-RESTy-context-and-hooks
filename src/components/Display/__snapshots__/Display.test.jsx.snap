// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component renders Display 1`] = `
<div
  className="Display"
>
  <t
    collapseStringsAfterLength={false}
    collapsed={false}
    defaultValue={null}
    displayDataTypes={true}
    displayObjectSize={true}
    enableClipboard={true}
    groupArraysAfterLength={100}
    iconStyle="triangle"
    indentWidth={4}
    name="Headers"
    onAdd={false}
    onDelete={false}
    onEdit={false}
    onSelect={false}
    shouldCollapse={false}
    sortKeys={false}
    src={Object {}}
    style={Object {}}
    theme="monokai"
    validationMessage="Validation Error"
  />
  <t
    collapseStringsAfterLength={false}
    collapsed={false}
    defaultValue={null}
    displayDataTypes={true}
    displayObjectSize={true}
    enableClipboard={true}
    groupArraysAfterLength={100}
    iconStyle="triangle"
    indentWidth={4}
    name="Response"
    onAdd={false}
    onDelete={false}
    onEdit={false}
    onSelect={false}
    shouldCollapse={false}
    sortKeys={false}
    src={Object {}}
    style={Object {}}
    theme="monokai"
    validationMessage="Validation Error"
  />
</div>
`;
